astropy/nddata/nddata.py

```diff
@@ -180,10 +196,21 @@ def __repr__(self):
 
     @property
     def data(self):
+        """
+        `~numpy.ndarray`: the data.
```

Comment:
It is definitely good to add a doscstring here -- but the return type will be ndarray-like, not necessarily ndarray.